CDP4Orm/AutoGenOrmHelpers/DaoResolver.cs

```diff
@@ -0,0 +1,226 @@
+// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="DaoResolver.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2024 RHEA System S.A.
+// 
+//    Author: Sam Gerené, Alex Vorobiev, Alexander van Delft, Nathanael Smiechowski, Antoine Théate
+// 
+//    This file is part of CDP4-COMET Webservices Community Edition.
+//    The CDP4-COMET Web Services Community Edition is the RHEA implementation of ECSS-E-TM-10-25 Annex A and Annex C.
+//    This is an auto-generated class. Any manual changes to this file will be overwritten!
+// 
+//    The CDP4-COMET Web Services Community Edition is free software; you can redistribute it and/or
+//    modify it under the terms of the GNU Affero General Public
+//    License as published by the Free Software Foundation; either
+//    version 3 of the License, or (at your option) any later version.
+// 
+//    The CDP4-COMET Web Services Community Edition is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+//    Lesser General Public License for more details.
+// 
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+// ------------------------------------------------------------------------------------------------
+// --------THIS IS AN AUTOMATICALLY GENERATED FILE. ANY MANUAL CHANGES WILL BE OVERWRITTEN!--------
+// ------------------------------------------------------------------------------------------------
+
+namespace CDP4Orm.Helper
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+
+    using CDP4Orm.Dao;
+
+    using Microsoft.Extensions.DependencyInjection;
+
+    /// <summary>
+    /// The <see cref="DaoResolver" /> provides resolve capabilites to retrieve <see cref="IDao" /> based on the type name
+    /// </summary>
+    public class DaoResolver: IDaoResolver
+    {
+        /// <summary>
+        /// Gets the <see cref="Dictionary{TKey,TValue}" /> that can provides
+        /// </summary>
+        private readonly Dictionary<string, Func<IDao>> existingDaos;
+
+        /// <summary>
+        /// Initializes a new <see cref="DaoResolver" />
+        /// </summary>
+        /// <param name="serviceProvider">The injected <see cref="IServiceProvider" /></param>
+        public DaoResolver(IServiceProvider serviceProvider)
```

Comment:
using serviceprovider.GetService is an anti-pattern https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines

i propose that instead we inject them as properties these services and that there is a helper function that returns the value of the property

Reply:
Should we inject them with a Lazy process? otherwise, at anytime we request DaoResolver, it would also initialize all Dao, no? With the IServiceProvider, we only request them when invoking the method?